ocl/oclapi/middlewares.py

```diff
@@ -63,3 +63,22 @@ def chunked_to_max(self, msg):
             return "{0}\n...\n".format(msg[0:MAX_BODY_LENGTH])
         else:
             return msg
+
+
+class FixMalformedLimitParamMiddleware(object):
+    """
+    Why this was necessary: https://github.com/OpenConceptLab/ocl_issues/issues/151
+    """
+    @staticmethod
+    def process_request(request):
+        to_remove = '?limit=100'
+        add_limit = False
+        if request.get_full_path()[-10:] == to_remove and request.method == 'GET':
+            query_dict_copy = request.GET.copy()
+            for key, value in query_dict_copy.iteritems():
+                if to_remove in value:
+                    query_dict_copy[key] = value.replace(to_remove, '')
+                    add_limit = True
+            if add_limit and 'limit' not in query_dict_copy:
```

Comment:
add_limit variable seems to be unnecessary... You already know that '?limit=100' was in the path.

Reply:
Wouldn't removing this mean we'd add the limit even to urls which didn't have it at this point?
Removing this would also allow us to remove the `if to_remove in value` so the two go together.

Reply:
`if request.get_full_path()[-10:] == to_remove` already tests, if the limit is there.

Reply:
Yes, but sometimes that is ok and we don't actually remove anything from any of the query dict entries.
Take a scenario where the request is `GET /concepts` and the limit is appended, giving `GET /concepts?limit=100`. This one would be captured correctly and we wouldn't need to fix any of the params.

Reply:
I understand that in case of /concepts?limit=100 the `if 'limit' not in query_dict_copy` would be false.

Reply:
+1 valid point. Removed the necessary variable and condition. Thanks for the catch.